person-directory-impl/src/main/java/org/apereo/services/persondir/util/CollectionsUtil.java

```diff
@@ -0,0 +1,41 @@
+package org.apereo.services.persondir.util;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Utility class to avoid nested Unmodifiable collections.
```

Comment:
It might be good to also add a few notes here, explaining the need for this class and the overall change.